diff --git a/node_modules/alias-hq/src/plugins/webpack/index.js b/node_modules/alias-hq/src/plugins/webpack/index.js
index b4c7e7e..6cbc6d2 100644
--- a/node_modules/alias-hq/src/plugins/webpack/index.js
+++ b/node_modules/alias-hq/src/plugins/webpack/index.js
@@ -1,11 +1,20 @@
 const { toObject, resolve } = require('../../utils')
 
+const defaults = {
+  format: 'string'
+}
+
 // @see https://webpack.js.org/configuration/resolve/#resolvealias
-function callback (name, [path], config) {
+function callback (name, paths, config, options) {
   const { root, base } = config
   name = name.replace(/\/\*$/, '')
-  path = path.replace(/\*$/, '')
-  path = resolve(root, base, path)
+  let path = paths.map(path => {
+    path = path.replace(/\*$/, '')
+    return resolve(root, base, path)
+  })
+  if (options.format === 'string' || path.length === 1) {
+    path = path[0]
+  }
   return {
     name,
     path,
@@ -13,5 +22,6 @@ function callback (name, [path], config) {
 }
 
 module.exports = function (config, options) {
+  options = { ...defaults, ...options }
   return toObject(callback, config, options)
 }
